### PR TITLE
Update mtenn pin for upcoming breaking release.

### PR DESCRIPTION
**Pull request by kaminow**
Going to cut an `mtenn` release soon that will break the current code, so pin to <= current version. Code is being updated to work with the new `mtenn` version in #1081.

### CODE REVIEW

File: devtools/conda-envs/asapdiscovery-macOS-12.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ dependencies:
   - dgl <=2.0.0
   - dgllife
   - pooch
-  - mtenn >=0.5.1
+  - mtenn >=0.5.1,<=0.5.2
```
Comment (ijpulidos) [inline devtools/conda-envs/asapdiscovery-macOS-12.yml:67]:
```suggestion
  - mtenn =0.5.1
```

File: devtools/conda-envs/asapdiscovery-macOS-latest.yml [hunk 2]
```diff
@@ -63,7 +63,7 @@ dependencies:
   - dgl <=2.0.0
   - dgllife
   - pooch
-  - mtenn >=0.5.1
+  - mtenn >=0.5.1,<=0.5.2
```
Comment (ijpulidos) [inline devtools/conda-envs/asapdiscovery-macOS-latest.yml:66]:
```suggestion
  - mtenn =0.5.1
```

File: devtools/conda-envs/asapdiscovery-ubuntu-latest.yml [hunk 3]
```diff
@@ -64,7 +64,7 @@ dependencies:
   - dgl <=2.0.0
   - dgllife
   - pooch
-  - mtenn >=0.5.1
+  - mtenn >=0.5.1,<=0.5.2
```
Comment (ijpulidos) [inline devtools/conda-envs/asapdiscovery-ubuntu-latest.yml:67]:
```suggestion
  - mtenn =0.5.1
```

**Review by ijpulidos — COMMENTED** [review]
Made some minor suggestions. Wouldn't it be better in terms of versioning for the new `mtenn` release to be `0.6.x`? I know we are in the `0.x` branch so it is not that important, but I think it would make it a little easier for the users to expect that `0.6` breaks compatibility with `0.5`. If that makes any sense.

**Comment by mikemhenry** [comment]
+1 to doing a release where the next version of mtenn is 0.6 and doing a `=` pin since we know there are no more 0.5.x releases and with the changes in mtenn we will need to do a code update

**Comment by kaminow** [comment]
that makes sense, thanks! updated `mtenn` pin to `=0.5.2` (most recent version on `conda-forge`)

**Review by mikemhenry — APPROVED** [review]

**Pull request merged**